src/Tests/Msbuild.Tests.Utilities/MockEngine.cs

```diff
@@ -0,0 +1,494 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Execution;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Logging;
+
+using FluentAssertions;
+using Xunit.Abstractions;
+
+#nullable disable
+
+namespace Msbuild.Tests.Utilities
+{
+    /***************************************************************************
+     * 
+     * Class:       MockEngine
+     * 
+     * In order to execute tasks, we have to pass in an Engine object, so the
+     * task can log events.  It doesn't have to be the real Engine object, just
+     * something that implements the IBuildEngine4 interface.  So, we mock up
+     * a fake engine object here, so we're able to execute tasks from the unit tests.
+     * 
+     * The unit tests could have instantiated the real Engine object, but then
+     * we would have had to take a reference onto the Microsoft.Build.Engine assembly, which
+     * is somewhat of a no-no for task assemblies.
+     * 
+     **************************************************************************/
+    public sealed class MockEngine : IBuildEngine7
```

Comment:
Would the existing [`MockBuildEngine`](https://github.com/dotnet/sdk/blob/main/src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/Mocks/MockBuildEngine.cs) class work here?  If not, we should probably update that class rather than introducing an entirely new mock engine and logger class.

Reply:
The built-in was missing implementations for IBuildEngine4 for general use (and 5 in a few test cases), so I did that and extracted the various subs in a couple other tests into this centralized version.